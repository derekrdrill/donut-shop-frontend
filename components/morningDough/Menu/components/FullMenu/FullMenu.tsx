import * as React from 'react';

import FullMenuHead from './components/FullMenuHead';
import FullMenuBody from './components/FullMenuBody';

import FULL_MENU_CATEGORIES from '../../assets/data/FULL_MENU_CATEGORIES';
import FULL_MENU from '../../assets/data/FULL_MENU';

const FullMenu = () => (
  <>
    <FullMenuHead />
    <FullMenuBody fullMenu={FULL_MENU} fullMenuCategories={FULL_MENU_CATEGORIES} />
  </>
);

export default FullMenu;
