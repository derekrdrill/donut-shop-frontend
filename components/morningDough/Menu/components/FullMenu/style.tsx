import { Grid, Typography } from '@mui/material';
import styled from 'styled-components';

export const FullMenuHeadContainer = styled(Grid)({
  backgroundColor: '#fef3f1',
  minHeight: 250,
  opacity: 0.8,
});

export const FullMenuHeadTextItem = styled(Grid)({
  padding: 20,
});

export const FullMenuContainer = styled(Grid)({
  paddingBottom: 40,
  margin: '40px 4% 0px 4%',
});

export const FullMenuText = styled(Typography)<{
  $fontColor?: string;
  $isSubtext?: boolean;
}>(({ $fontColor, $isSubtext }) => [
  {
    color: $fontColor,
    fontFamily: `'Lilita One', cursive !important`,
    margin: '15px 0',
  },
  $isSubtext && {
    fontFamily: `'Rubik', sans-serif !important`,
  },
]);

export const FullMenuHeadImage = styled.img({
  height: 225,
  width: 225,
});

export const FullMenuItemRoot = styled(Grid)({
  height: 200,
  padding: 10,
});

export const FullMenuItemContainer = styled(Grid)({
  ':hover': {
    backgroundColor: '#f8d7ff',
  },
  border: '1px solid #ababab',
  borderRadius: 10,
  cursor: 'pointer',
  maxWidth: 340,
  transition: 'all 250ms ease-in',
});

export const FullMenuItemImage = styled.img({
  height: 150,
  width: 150,
});
