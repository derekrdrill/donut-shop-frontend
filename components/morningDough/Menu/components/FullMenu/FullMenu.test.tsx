jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import FullMenu from './FullMenu';

describe('FullMenu unit tests', () => {
  it('renders as expected', () => {
    const result = renderer.create(<FullMenu />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
