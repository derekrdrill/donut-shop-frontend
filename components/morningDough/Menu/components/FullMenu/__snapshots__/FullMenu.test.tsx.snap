// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullMenu unit tests renders as expected 1`] = `
[
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-beqWaB hTrTut css-11lq3yg-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-1aljnjw-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-68owf2-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1079s95-MuiGrid-root"
            >
              <img
                className="sc-eDDNvR gZzQUy"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1xf4ifq-MuiGrid-root"
            >
              <img
                className="sc-eDDNvR gZzQUy"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 sc-gueYoa jmZJof css-mb590u-MuiGrid-root"
        >
          <h2
            className="MuiTypography-root MuiTypography-h2 sc-hLseeU PzYec css-10sp2wv-MuiTypography-root"
          >
            FIRE UP THAT ORDER ðŸ”¥
          </h2>
          <h3
            className="MuiTypography-root MuiTypography-h3 sc-hLseeU hJUVBK css-1pm15ke-MuiTypography-root"
          >
            We'll be ready ðŸ˜Ž
          </h3>
          <h6
            className="MuiTypography-root MuiTypography-h6 sc-hLseeU cHiVpT css-1u98osu-MuiTypography-root"
          >
            Special tasty treats just for you
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body2 sc-hLseeU bpCwal css-odxzrf-MuiTypography-root"
          >
            $1 croissants or FREE small coffee with the pruchase of an avacado toast. Price and participation may vary. Limited time offer. Terms apply.
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-ax2l3g-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-sbmkh9-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1079s95-MuiGrid-root"
            >
              <img
                className="sc-eDDNvR gZzQUy"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-12 MuiGrid-grid-lg-6 css-1xf4ifq-MuiGrid-root"
            >
              <img
                className="sc-eDDNvR gZzQUy"
                src={
                  {
                    "process": [Function],
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container sc-dmqHEX hMzWU css-851zx1-MuiGrid-root"
    data-testid="full-menu"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
      >
        Full Menu
      </h4>
      <div
        className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        data-testid="full-menu-back-button"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-iAEyYk lhGoHL"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
          >
            Hot Drinks
          </h6>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-iAEyYk lhGoHL"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
          >
            Cold Drinks
          </h6>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-iAEyYk lhGoHL"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
          >
            Donuts
          </h6>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-iAEyYk lhGoHL"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
          >
            Bagels
          </h6>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-iAEyYk lhGoHL"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
          >
            Muffins
          </h6>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-iAEyYk lhGoHL"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
          >
            Croissants
          </h6>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
        onClick={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-iAEyYk lhGoHL"
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
          >
            Toast
          </h6>
        </div>
      </div>
    </div>
  </div>,
]
`;
